flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/TimeGaugeTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// ----------------------------------------------------------------------------
+//  This class is largely adapted from "com.google.common.base.Preconditions",
+//  which is part of the "Guava" library.
+//
+//  Because of frequent issues with dependency conflicts, this class was
+//  added to the Flink code base to reduce dependency on Guava.
+// ----------------------------------------------------------------------------
+
+package org.apache.flink.runtime.metrics.groups;
+
+import org.apache.flink.metrics.View;
+
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/** Tests for {@link TimeGauge}. */
+public class TimeGaugeTest {
+    private static final long SLEEP = 10;
+
+    @Test
+    public void testBasicUsage() throws Exception {
+        TimeGauge gauge = new TimeGauge();
+
+        gauge.update();
+        assertThat(gauge.getValue(), is(0L));
+
+        gauge.markStart();
+        Thread.sleep(SLEEP);
```

Comment:
You could make testing easier by making the time measurements be based on a `Clock`.